nym-vpn-apple/Services/Sources/Services/NymLogger/LogFileManager.swift

```diff
@@ -24,46 +34,61 @@ public final class LogFileManager {
         guard let logsDirectory else { return nil }
 
         try? fileManager.createDirectory(at: logsDirectory, withIntermediateDirectories: true, attributes: nil)
-        let logFileURL = logsDirectory.appendingPathComponent(Constants.logFileName.rawValue)
+        let fileName = "\(logFileType.rawValue)\(Constants.logFileName.rawValue)"
+        let logFileURL = logsDirectory.appendingPathComponent(fileName)
+
         return logFileURL
     }
 
-    public func logs() -> String {
-        guard let logFileURL = logFileURL,
-              let logData = try? Data(contentsOf: logFileURL),
-              let appLogs = String(data: logData, encoding: .utf8)
-        else {
-            return ""
+    public func write(_ string: String) {
+        ioQueue.async {
+            try? self.fileHandle?.write(contentsOf: Data(string.utf8))
         }
-        return appLogs
     }
 
     public func deleteLogs() {
         ioQueue.async {
-            guard let logFileURL = self.logFileURL else { return }
+            guard let logFileURL = self.logFileURL(logFileType: self.logFileType) else { return }
+            if let appLogFileURL = self.logFileURL(logFileType: .app) {
+                try? FileManager.default.removeItem(at: appLogFileURL)
+            }
+            if let tunnelLogFileURL = self.logFileURL(logFileType: .tunnel) {
+                try? FileManager.default.removeItem(at: tunnelLogFileURL)
+            }
+            try? self.fileHandle?.close()
             try? FileManager.default.removeItem(at: logFileURL)
             self.fileHandle = nil
-        }
-    }
 
-    public func write(_ string: String) {
-        ioQueue.async {
-            try? self.fileHandle?.write(contentsOf: Data(string.utf8))
+            DarwinNotificationCenter.shared.post(name: DarwinNotificationKey.reconfigureLogs.key)
         }
     }
 }
 
 private extension LogFileManager {
+    func setup() {
+        notificationObservation = DarwinNotificationCenter.shared.addObserver(
+            name: DarwinNotificationKey.reconfigureLogs.key
+        ) { [weak self] in
+            self?.fileHandle = nil
```

Comment:
Since you mutate `self.fileHandle` in here you need it wrapped into `ioQueue.async { }` to avoid out of order access to the member variable.